lib/datadog/tracing/contrib/rack/middlewares.rb

```diff
@@ -304,7 +304,15 @@ def parse_url(env, original_env)
                        else
                          # normally REQUEST_URI starts at the path, but it
                          # might contain the full URL in some cases (e.g WEBrick)
-                         request_uri.sub(/^#{base_url}/, '')
+                         #
+                         # DEV: Use `request_uri.delete_prefix(base_url)`, supported in Ruby 2.5+
+                         # rubocop:disable Style/IfInsideElse
+                         if request_uri.rindex(base_url, 0)
```

Comment:
Minor: I'm curious, is `.rindex(base_url, 0)` an optimization vs `start_with?`?

Reply:
I think I got this from an old Ruby pattern, but I just tested and `start_with?` is faster (and more legible), so I changed it.
Thank you!